project/JavalibIRCleaner.scala

```diff
@@ -0,0 +1,404 @@
+          tree
+      }
+
+      val result = super.transform(preprocessedTree, isStat) match {
+        case New(cls, ctor, args) =>
+          New(cls, transformMethodIdent(ctor), args)
+
+        case t: Apply =>
+          Apply(t.flags, t.receiver, transformMethodIdent(t.method),
+              t.args)(t.tpe)
+        case t:ApplyStatically =>
+          ApplyStatically(t.flags, t.receiver, transformClassRef(t.cls),
+              transformMethodIdent(t.method), t.args)(t.tpe)
+        case t: ApplyStatic =>
+          ApplyStatic(t.flags, transformClassRef(t.cls),
+              transformMethodIdent(t.method), t.args)(t.tpe)
+
+        case NewArray(typeRef, lengths) =>
+          NewArray(transformArrayTypeRef(typeRef), lengths)
+        case ArrayValue(typeRef, elems) =>
+          ArrayValue(transformArrayTypeRef(typeRef), elems)
+
+        case IsInstanceOf(expr, typeRef) =>
+          IsInstanceOf(expr, transformTypeRef(typeRef))
+        case AsInstanceOf(expr, typeRef) =>
+          AsInstanceOf(expr, transformTypeRef(typeRef))
+
+        case LoadJSConstructor(cls) =>
+          genLoadFromLoadSpecOf(cls)
+        case LoadJSModule(cls) =>
+          genLoadFromLoadSpecOf(cls)
+
+        case t: ClassOf =>
+          if (transformTypeRef(t.typeRef) != t.typeRef)
+            System.err.println(s"illegal Class(${t.typeRef}) at $pos")
+          t
+
+        case t =>
+          t
+      }
+
+      validateType(result.tpe)
+      result
+    }
+
+    private def genLoadFromLoadSpecOf(cls: ClassRef)(
+        implicit pos: Position): Tree = {
+      jsTypes.get(cls.className) match {
+        case Some(classDef) =>
+          classDef.jsNativeLoadSpec match {
+            case Some(loadSpec) =>
+              genLoadFromLoadSpec(loadSpec)
+            case None =>
+              System.err.println(s"$cls does not have a load spec at $pos")
+              JSGlobalRef(Ident("Object"))
+          }
+        case None =>
+          System.err.println(s"$cls is not a JS type at $pos")
+          JSGlobalRef(Ident("Object"))
+      }
+    }
+
+    private def genLoadFromLoadSpec(loadSpec: JSNativeLoadSpec)(
+        implicit pos: Position): Tree = {
+      loadSpec match {
+        case JSNativeLoadSpec.Global(globalRef, Nil) =>
+          JSGlobalRef(Ident(globalRef))
+        case _ =>
+          System.err.println(s"unsupported load spec $loadSpec at $pos")
+          JSGlobalRef(Ident("Object"))
+      }
+    }
+
+    private def transformMethodIdent(ident: Ident): Ident = {
+      implicit val pos = ident.pos
+      val encodedName = ident.encodedName
+      val sig = decodeMethodName(encodedName)
+      val newParamTypes = sig._2.map(transformTypeRef)
+      val newResultType = sig._3.map(transformTypeRef)
+      if (newParamTypes == sig._2 && newResultType == sig._3) {
+        ident
+      } else {
+        Ident(encodeMethodName(sig._1, newParamTypes, newResultType),
+            ident.originalName)(ident.pos)
+      }
+    }
```

Comment:
How do you make sure this does not generate two methods with the same signature?

Reply:
I have added code in `cleanClassDef` to check that.